catalog/rest/rest.go

```diff
@@ -221,12 +222,12 @@ func (s *sessionTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 				return nil, err
 			}
 
-			if _, err = io.Copy(s.h, rdr); err != nil {
+			h := s.newHash()
+			if _, err = io.Copy(h, rdr); err != nil {
 				return nil, err
 			}
 
-			payloadHash = string(s.h.Sum(nil))
-			s.h.Reset()
+			payloadHash = hex.EncodeToString(h.Sum(nil))
```

Comment:
Oops -- really need an integration test of some sort for this. Unfortunately, the AWS Go library doesn't seem to include a signature verifier that we can easily drop into a test server to make sure these are correct. But after stepping through the new test, I _think_ it should all work, with these fixes in.